### PR TITLE
[MXNET-347] Logical AND operator

### PR DESCRIPTION
**Pull request by anirudhacharya**
## Description ##
Logical AND operator

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Logical ANd op.

## Comments ##
- @haojin2 @eric-haibin-lin

### CODE REVIEW

File: tests/python/unittest/test_operator.py [hunk 1]
```diff
@@ -1564,6 +1564,13 @@ def test_bmin(a, b):
         # pass idx=200 to gen_broadcast_data so that generated ndarrays' sizes are not too big
         data = gen_broadcast_data(idx=200)
         check_bmaxmin_gradient(c, data[0], data[1], 0.001, 1e-2, 1e-3)
+        
+    def test_band(a, b):
+        c = mx.sym.broadcast_logical_and(a, b)
+        check_binary_op_forward(c, lambda x, y: np.logical_and(x, y), gen_broadcast_data, mx_nd_func=mx.nd.logical_and)
+        # pass idx=200 to gen_broadcast_data so that generated ndarrays' sizes are not too big
+        data = gen_broadcast_data(idx=200)
+        check_bmaxmin_gradient(c, data[0], data[1], 0.001, 1e-2, 1e-3)
```
Comment (haojin2) [inline tests/python/unittest/test_operator.py]: Is this a proper gradient check for this op?

Comment (anirudhacharya) [reply tests/python/unittest/test_operator.py]: no its not. let me push the changes.

File: src/operator/tensor/elemwise_binary_broadcast_op_logic.cu [hunk 2]
```diff
@@ -47,5 +47,9 @@ NNVM_REGISTER_OP(broadcast_lesser)
 NNVM_REGISTER_OP(broadcast_lesser_equal)
 .set_attr<FCompute>("FCompute<gpu>", BinaryBroadcastCompute<gpu, mshadow_op::le>);
 
+// logical and
+NNVM_REGISTER_OP(logical_and)
```
Comment (anirudhacharya) [inline src/operator/tensor/elemwise_binary_broadcast_op_logic.cu]: should change this to broadcast_logical_and

File: src/operator/tensor/elemwise_binary_broadcast_op_logic.cu [hunk 3]
```diff
@@ -47,5 +47,9 @@ NNVM_REGISTER_OP(broadcast_lesser)
 NNVM_REGISTER_OP(broadcast_lesser_equal)
 .set_attr<FCompute>("FCompute<gpu>", BinaryBroadcastCompute<gpu, mshadow_op::le>);
 
+// logical and
+NNVM_REGISTER_OP(broadcast_logical_and)
+.set_attr<FCompute>("FCompute<gpu>", BinaryBroadcastCompute::Compute<gpu, mshadow_op::logical_and>);
```
Comment (haojin2) [inline src/operator/tensor/elemwise_binary_broadcast_op_logic.cu]: should be BinaryBroadcastCompute<gpu, mshadow_op::logical_and>

**Comment by anirudhacharya** [comment]
part of this PR - https://github.com/apache/incubator-mxnet/pull/10679

**Pull request closed without merging**